repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -375,16 +383,56 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
     * Converts the saml response document into a servlet input stream for passing down the filter chain.
     *
     * @param document the final saml response
-    * @return the sinput stream that can be wrapped and passed along
+    * @return the input stream that can be wrapped and passed along
     */
-  def convertDocumentToStream(document: Document): ServletInputStream = ???
+  def convertDocumentToStream(document: Document): ServletInputStream = {
+    val source = new DOMSource(document)
+    val baos = new ByteArrayOutputStream()
+    val outputTarget = new StreamResult(baos)
+    transformerFactory.newTransformer().transform(source, outputTarget)
+    new BufferedServletInputStream(new ByteArrayInputStream(baos.toByteArray))
+  }
 
   /**
-    * This is super vague because we don't know what it looks like yet.
+    * Added the extended attributes passed in the SAML Response.
     *
     * @return
     */
-  def doResponseStuff() = ???
+  def addExtendedAttributes(response: HttpServletResponseWrapper, translatedSamlResponse: Document): Option[InputStream] = {
+    logger.debug("Starting the Response processing.")
+    response.getContentType.toLowerCase match {
+      case json if json.contains("json") =>
+        val baos = new ByteArrayOutputStream
+        val destination = AttributeMapper.processor.newSerializer(baos)
+        AttributeMapper.addExtendedAttributes(
+          new StreamSource(response.getOutputStreamAsInputStream),
+          new StreamSource(convertDocumentToStream(translatedSamlResponse)),
```

Comment:
There is not one that exactly matched what we had, but after looking again there is one that matches a little better. At least using a couple of the available helper methods.